js/src/sdk/models/triggers.ts

```diff
@@ -29,18 +40,31 @@ export class Triggers {
    * @returns {CancelablePromise<ListTriggersResponse>} A promise that resolves to the list of all triggers.
    * @throws {ApiError} If the request fails.
    */
-  async list(data: RequiredQuery = {}) {
+  async list(data: TTriggerQuery = {}) {
     TELEMETRY_LOGGER.manualTelemetry(TELEMETRY_EVENTS.SDK_METHOD_INVOKED, {
       method: "list",
       file: this.fileName,
       params: { data },
     });
+    const {appNames, triggerIds, connectedAccountsIds, integrationIds, showEnabledOnly} = ZTriggerQuery.parse(data);
+   
+    if(!appNames && !triggerIds && !connectedAccountsIds && !integrationIds) {
```

Comment:
Consider adding validation for empty arrays as well. Currently, passing an empty array would pass this validation but might not be a valid use case. You could add:

```typescript
const hasValidFilter = (arr?: string[]) => arr && arr.length > 0;
if (!hasValidFilter(appNames) && !hasValidFilter(triggerIds) && 
    !hasValidFilter(connectedAccountsIds) && !hasValidFilter(integrationIds)) {
  throw CEG.getCustomError(...);
}
```